File: VB/E1984/SerializableFormatCondition.vb
Imports DevExpress.Utils
Imports DevExpress.Utils.Serializing
Imports DevExpress.XtraGrid
Imports DevExpress.XtraTreeList.StyleFormatConditions

Namespace E1984
	Public Class SerializableFormatCondition

		Private _appearance As New AppearanceObject()

		Private _applyToRow As Boolean

		Private _columnName As String = ""

		Private _condition As FormatConditionEnum

		Private _value1 As Object

		Private _value2 As Object

		Private _expression As Object


		<XtraSerializableProperty(XtraSerializationVisibility.Content)>
		Public Property Appearance() As AppearanceObject
			Get
				Return _appearance
			End Get
			Set(ByVal value As AppearanceObject)
				_appearance.Assign(value)
			End Set
		End Property

		<XtraSerializableProperty()>
		Public Property ColumnName() As String
			Get
				Return _columnName
			End Get
			Set(ByVal value As String)
				_columnName = value
			End Set
		End Property

		<XtraSerializableProperty()>
		Public Property Condition() As FormatConditionEnum
			Get
				Return _condition
			End Get
			Set(ByVal value As FormatConditionEnum)
				_condition = value
			End Set
		End Property

		<XtraSerializableProperty()>
		Public Property ApplyToRow() As Boolean
			Get
				Return _applyToRow
			End Get
			Set(ByVal value As Boolean)
				_applyToRow = value
			End Set
		End Property

		<XtraSerializableProperty()>
		Public Property Value1() As Object
			Get
				Return _value1
			End Get
			Set(ByVal value As Object)
				_value1 = value
			End Set
		End Property

		<XtraSerializableProperty()>
		Public Property Value2() As Object
			Get
				Return _value2
			End Get
			Set(ByVal value As Object)
				_value2 = value
			End Set
		End Property
		<XtraSerializableProperty()>
		Public Property Expression() As Object
			Get
				Return _expression
			End Get
			Set(ByVal value As Object)
				_expression = value
			End Set
		End Property
		Public Sub CopyFromStyleFormat(ByVal style As StyleFormatCondition)
			Appearance = style.Appearance
			ApplyToRow = style.ApplyToRow
			ColumnName = If(style.Column Is Nothing, String.Empty, style.Column.Name)
			Condition = style.Condition
			Value1 = style.Value1
			Value2 = style.Value2
			Expression = style.Expression
		End Sub

		Public Sub CopyToStyleFormat(ByVal style As StyleFormatCondition)
			style.Appearance.Assign(Appearance)
			style.ApplyToRow = ApplyToRow
			style.Column = CType(style.Collection, StyleFormatConditionCollection).TreeListControl.Columns.ColumnByName(ColumnName)
			style.Condition = Condition
			style.Value1 = Value1
			style.Value2 = Value2
			style.Expression = Expression.ToString()
		End Sub
	End Class
End Namespace

File: CS/E1984/SerializableFormatCondition.cs
using DevExpress.Utils;
using DevExpress.Utils.Serializing;
using DevExpress.XtraGrid;
using DevExpress.XtraTreeList.StyleFormatConditions;

namespace E1984 {
    public class SerializableFormatCondition
    {

        private AppearanceObject _appearance = new AppearanceObject();

        private bool _applyToRow;

        private string _columnName = "";

        private FormatConditionEnum _condition;

        private object _value1;

        private object _value2;

        private object _expression;


        [XtraSerializableProperty(XtraSerializationVisibility.Content)]
        public AppearanceObject Appearance {
            get => _appearance;
            set => _appearance.Assign(value);
        }

        [XtraSerializableProperty()]
        public string ColumnName {
            get => _columnName;
            set => _columnName = value;
        }

        [XtraSerializableProperty()]
        public FormatConditionEnum Condition {
            get => _condition;
            set => _condition = value;
        }

        [XtraSerializableProperty()]
        public bool ApplyToRow {
            get => _applyToRow;
            set => _applyToRow = value;
        }

        [XtraSerializableProperty()]
        public object Value1 {
            get => _value1;
            set => _value1 = value;
        }

        [XtraSerializableProperty()]
        public object Value2 {
            get => _value2;
            set => _value2 = value;
        }
        [XtraSerializableProperty()]
        public object Expression {
            get => _expression;
            set => _expression = value;
        }
        public void CopyFromStyleFormat(StyleFormatCondition style)
        {
            Appearance = style.Appearance;
            ApplyToRow = style.ApplyToRow;
            ColumnName = style.Column == null? string.Empty:  style.Column.Name;
            Condition = style.Condition;
            Value1 = style.Value1;
            Value2 = style.Value2;
            Expression = style.Expression;
        }

        public void CopyToStyleFormat(StyleFormatCondition style)
        {
            style.Appearance.Assign(Appearance);
            style.ApplyToRow = ApplyToRow;
            style.Column = ((StyleFormatConditionCollection)style.Collection).TreeListControl.Columns.ColumnByName(ColumnName);
            style.Condition = Condition;
            style.Value1 = Value1;
            style.Value2 = Value2;
            style.Expression = Expression.ToString();
        }
    }
}